### ISSUE TITLE
Feature: Add offline capabilities

### ISSUE THREAD

**Issue by wesleygrimes** [nestjs-addons/in-memory-db#4]
Would be nice to have the ability to save records on disk to JSON files so that on restart data would be available.

**Comment by wesleygrimes** [comment]
@cmwhited now that pr #21 has landed, I think it would be a fantastic addition to add offline capability. Did you want to take this one on? It's fairly siloed so you should be able to work on it as time permits without others duplicating efforts.

**Comment by wesleygrimes** [comment]
@cmwhited I will assign you, for now, feel free to change assignment if you want to work on something else.

**Comment by cmwhited** [comment]
Yep. I will rebase on master and bring these changes in. Makes it a bit easier for what I am doing for this functionality. Going to be working on this tonight/tomorrow. Hopefully, PR by tomorrow.

**Comment by cvramirez** [comment]
I just ran into your project on YouTube.  Nice!  👍 

FYI, here's a similar project to yours.  https://github.com/hungtcs/nest-nedb

Since it wraps NeDB, it's both in-memory with a persistence option.  I haven't tried it yet, but I've used NeDB on other non-nestjs projects, feathersjs actually, and found their NeDB adapter works quite well, even for hundreds of records, and the persistence part is great.  An option to think about just in case you want to keep your in-memory-db project in memory without scope creep.
